### PR TITLE
First Commit

### PR DESCRIPTION
**Pull request by cs-sujeet-ahirrao**
First Commit

### CODE REVIEW

File: connector.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Copyright (C) 2008 - 2021 Fortinet Inc. 
+All rights reserved. 
+FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE 
+Copyright end 
+""" 
+from connectors.core.connector import Connector, get_logger, ConnectorError
+
+from .operations import operations, _check_health
+
+logger = get_logger('anomali_taxii2_feed')
```
Comment (cs-pooja-singh) [inline connector.py:12]: anomali_limo_feed

File: info.json [hunk 2]
```diff
@@ -0,0 +1,271 @@
+{
+  "name": "anomali-limo-feed",
+  "label": "Anomali ThreatStream TAXII Feed",
```
Comment (cs-pooja-singh) [inline info.json:3]: Anomali Limo Intel Feed

File: info.json [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+  "description": "Anomali ThreatStream TAXII 2.0 Client",
```
Comment (cs-pooja-singh) [inline info.json:4]: Anomali Limo Threat Intelligence Feed Integration

File: connector.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class TaxiiFeedCon(Connector):
```
Comment (cs-pooja-singh) [inline connector.py:15]: AnomaliLimoFeed

File: connector.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    def execute(self, config, operation, params, **kwargs):
+        logger.info('In execute() Operation: {}'.format(operation))
```
Comment (cs-pooja-singh) [inline connector.py:17]: debug

File: info.json [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+  "publisher": "Fortinet",
+  "cs_approved": true,
+  "cs_compatible": true,
+  "version": "1.0.0",
+  "category": "SIEM",
```
Comment (cs-pooja-singh) [inline info.json:9]: Threat Intelligence

File: info.json [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+        "name": "verify_ssl",
+        "value": true
+      }
+    ]
+  },
+  "operations": [
+    {
+      "operation": "get_api_root_information",
+      "title": "Get API Root Information",
+      "description": "Returns information contains title, description, TAXII versions and maximum size of the content body.",
```
Comment (cs-pooja-singh) [inline info.json:57]: Returns information contains title, description, TAXII versions and maximum size of the content body for the specified TAXII server

File: info.json [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+        ]
+      },
+      "parameters": [
+      ],
+      "enabled": true
+    },
+    {
+      "operation": "get_collections",
+      "title": "Get Collections",
+      "description": "Returns a list of public/private/shared collection resources which users have the access.",
```
Comment (cs-pooja-singh) [inline info.json:74]: Returns the list of public/private/shared collection resources to which the user has access.

File: info.json [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+          "name": "added_after",
+          "visible": true,
+          "required": false,
+          "editable": true,
+          "type": "datetime",
+          "tooltip": "Filter the result set to only include the items added after the specified time.",
+          "description": "Filter the result set to only include the items added after the specified time."
+        },
+        {
+          "title": "Save Response in File",
```
Comment (cs-pooja-singh) [inline info.json:132]: Save Response to File

File: info.json [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+            "true": [
+              {
+                "title": "Filename",
+                "name": "filename",
+                "type": "text",
+                "visible": true,
+                "editable": true,
+                "required": false,
+                "placeholder": "Eg. demo.json",
+                "tooltip": "file will create under /tmp directory"
```
Comment (cs-pooja-singh) [inline info.json:148]: The file will be created under /tmp

File: info.json [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+              }
+            ]
+          },
+          "tooltip": "If check then save the response in File else in JSON",
```
Comment (cs-pooja-singh) [inline info.json:152]: If enabled, the feed will be written to a value, else returned as a JSON

File: info.json [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+          "type": "checkbox",
+          "onchange": {
+            "true": [
+              {
+                "title": "Filename",
+                "name": "filename",
+                "type": "text",
+                "visible": true,
+                "editable": true,
+                "required": false,
```
Comment (cs-pooja-singh) [inline info.json:146]: give it a default true. Writing to file should be default

File: info.json [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+          "tooltip": "If check then save the response in File else in JSON",
+          "description": "Specify the response field, If check then save the response in File else in JSON"
+        }
+      ],
+      "enabled": true
+    },
+    {
+      "operation": "get_output_schema",
+      "annotation": "get_output_schema",
+      "description": "Gives the output schema based on the inputs",
```
Comment (cs-pooja-singh) [inline info.json:161]: Generates output schema for other actions based on the input configuration

File: info.json [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+      "title": "Get Output Schema",
+      "enabled": true,
+      "visible": false,
+      "parameters": [],
+      "output_schema": {}
+    },
+    {
+      "operation": "get_objects_by_object_id",
+      "title": "Get Object By Object ID",
+      "description": "Returns general information for the specific object in one collection.",
```
Comment (cs-pooja-singh) [inline info.json:171]: of a collection

File: info.json [hunk 15]
```diff
@@ -0,0 +1,271 @@
...
+      },
+      "parameters": [
+        {
+          "title": "collection ID",
+          "name": "collectionID",
+          "visible": true,
+          "required": true,
+          "editable": true,
+          "type": "text",
+          "tooltip": "Returns general information about the collection, such as id, title, description, can_read, can_write and media_types.",
```
Comment (cs-pooja-singh) [inline info.json:212]: does this action return collection info?

File: operations.py [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE 
+Copyright end 
+""" 
+import base64
+
+import requests
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.cyops_utilities.builtins import create_file_from_string
+
+logger = get_logger('anomali_taxii2_feed')
```
Comment (cs-pooja-singh) [inline operations.py:14]: anomali_limo_feed

File: operations.py [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class TaxiiFeed(object):
```
Comment (cs-pooja-singh) [inline operations.py:17]: TaxiiClient

File: operations.py [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+    else:
+        return {'collections': [response]}
+
+
+def get_objects_by_collection_id(config, params):
+    taxii = TaxiiFeed(config)
+    params = get_params(params)
+    wanted_keys = set(['added_after'])
+    query_params = {k: params[k] for k in params.keys() & wanted_keys}
+    response = taxii.make_request_taxii(endpoint='collections/' + params['collectionID'] + '/objects/',
```
Comment (cs-pooja-singh) [inline operations.py:122]: does this work for data more than 1000? doesn't seem to do pagination correctly.
use a taxii client instead. eg:

server = Server('https://limo.anomali.com/api/v1/taxii2/taxii/', user='guest', password='guest')
api_root = server.api_roots[0]
for collection in api_root.collections:
      for bundle in as_pages(collection.get_objects, per_request=1000):

File: info.json [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+  "name": "anomali-limo-feed",
+  "label": "Anomali ThreatStream TAXII Feed",
+  "description": "Anomali ThreatStream TAXII 2.0 Client",
+  "publisher": "Fortinet",
+  "cs_approved": true,
+  "cs_compatible": true,
+  "version": "1.0.0",
+  "category": "SIEM",
+  "icon_small_name": "small_icon.png",
+  "icon_large_name": "large_icon.png",
```
Comment (cs-pooja-singh) [inline info.json:11]: include sample playbooks as well

File: info.json (hunk 15 above)
Comment (cs-sujeet-ahirrao) [reply info.json:212]: No,  It retrieves specific object information.
Changed tooltip and description tp "Specify the Collection ID, who's Object information you want to retrieve"

**Comment by cs-pooja-singh** [comment]
closing as a new PR with all review comments addressed was raised

**Pull request closed without merging**